phe/tests/paillier_test.py

```diff
@@ -193,63 +198,96 @@ def testEncodeIntDecodeInt4(self):
 
     def testEncodeIntDecodeInt5(self):
         # largest negative number
-        encoded = paillier.EncodedNumber.encode(self.public_key,
+        encoded = self.EncodedNumberCls.encode(self.public_key,
                                                 -self.public_key.max_int)
         decoded = encoded.decode()
         self.assertEqual(-self.public_key.max_int, decoded)
 
     def testEncodeIntTooLargePositive(self):
         # check value error is raised on too large a positive input
-        self.assertRaises(ValueError, paillier.EncodedNumber.encode,
+        self.assertRaises(ValueError, self.EncodedNumberCls.encode,
                           self.public_key, self.public_key.max_int + 1)
-        self.assertRaises(ValueError, paillier.EncodedNumber.encode,
+        self.assertRaises(ValueError, self.EncodedNumberCls.encode,
                           self.public_key, 2 ** (paillier.DEFAULT_KEYSIZE-1))
 
     def testEncodeIntTooLargeNegative(self):
         # check value error is raised on too large a positive input
-        self.assertRaises(ValueError, paillier.EncodedNumber.encode,
+        self.assertRaises(ValueError, self.EncodedNumberCls.encode,
                           self.public_key, -self.public_key.max_int - 1)
-        self.assertRaises(ValueError, paillier.EncodedNumber.encode,
+        self.assertRaises(ValueError, self.EncodedNumberCls.encode,
                           self.public_key, -2 ** (paillier.DEFAULT_KEYSIZE-1))
 
     def testDecodeCorruptEncodedNumber(self):
-        encoded = paillier.EncodedNumber.encode(self.public_key, 10)
+        encoded = self.EncodedNumberCls.encode(self.public_key, 10)
         encoded.encoding += self.public_key.n
         self.assertRaises(ValueError, encoded.decode)
 
     def testDecodeWithOverflowEncodedNumber(self):
-        encoded = paillier.EncodedNumber.encode(self.public_key, 10)
+        encoded = self.EncodedNumberCls.encode(self.public_key, 10)
         encoded.encoding += self.public_key.max_int
         self.assertRaises(OverflowError, encoded.decode)
 
     def testEncodeFloat0(self):
-        enc = paillier.EncodedNumber.encode(self.public_key, 15.1)
-        negexp = paillier.EncodedNumber.BASE ** enc.exponent
-        dec = paillier.EncodedNumber.BASE ** enc.exponent * enc.encoding
+        enc = self.EncodedNumberCls.encode(self.public_key, 15.1)
+        negexp = self.EncodedNumberCls.BASE ** enc.exponent
+        dec = self.EncodedNumberCls.BASE ** enc.exponent * enc.encoding
         self.assertEqual(15.1, dec)
 
     def testEncodeFloatDecodeFloat0(self):
-        enc = paillier.EncodedNumber.encode(self.public_key, 15.1)
+        enc = self.EncodedNumberCls.encode(self.public_key, 15.1)
         self.assertEqual(15.1, enc.decode())
```

Comment:
Might want to check whether this still works when BASE != 2**n... might have to slip in an `assertAlmostEqual` if you want to allow arbitrary BASEs